.github/workflows/composer_install.yml

```diff
@@ -0,0 +1,34 @@
+name: Composer Install
+
+on:
+    push:
+        branches: [ master ]
+    pull_request:
+
+env:
+    # see https://github.com/composer/composer/issues/9368#issuecomment-718112361
+    # see https://getcomposer.org/doc/articles/troubleshooting.md#dependencies-on-the-root-package
+    COMPOSER_ROOT_VERSION: 'dev-master'
+
+jobs:
+    composer_install:
+        strategy:
+            fail-fast: false
+            matrix:
+                php_version:
+                    - 7.4
+                    - 8.0
+
+        name: composer install on PHP ${{ matrix.package }}
+        runs-on: ubuntu-latest
+        steps:
+            -   uses: actions/checkout@v2
+
+            -
+                name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: ${{ matrix.package }}
+
+            -
+                uses: "ramsey/composer-install@v1"
```

Comment:
See https://github.com/ramsey/composer-install#usage

This should simplify the other jobs, as this action handles caching :muscle: 

![image](https://user-images.githubusercontent.com/924196/128902998-25535cf4-097f-4abf-8ede-aa5845be5e0d.png)